erpnext/stock/doctype/item/item.py

```diff
@@ -344,6 +315,17 @@ def validate_stock_for_template_must_be_zero(self):
 				or ifnull(reserved_qty, 0) > 0 or ifnull(indented_qty, 0) > 0 or ifnull(planned_qty, 0) > 0)""", self.name)
 			if stock_in:
 				frappe.throw(_("Item Template cannot have stock or Open Sales/Purchase/Production Orders."), ItemTemplateCannotHaveStock)
+				
+	def validate_uom(self):
+		if not self.get("__islocal"):
+			check_stock_uom_with_bin(self.item_code, self.stock_uom)
```

Comment:
Never check with self.item_code, use self.name

name can be different from item code, there is a setting in Stock Settings